### PR TITLE
add app protocol to http builder

### PR DESCRIPTION
**Pull request by OfTheDelmer**
Resolves https://github.com/ngrok/ngrok-javascript/issues/107

### Why

To allow specifying `http1` or `http2` for listeners

### How

Add `app_protocol` to both `HTTPListenerBuilder` and `config`.

Add an example with node's http2 example.

### Validation

Run example with local build and can view `Hello World` in browser on listener url.

### CODE REVIEW

**Comment by OfTheDelmer** [comment]
Note: I didn't add an Express example because Express 5 and http2 support is still in beta. 😂 
https://expressjs.com/en/5x/api.html

File: examples/ngrok-http2.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+const ngrok = require("../index.js");
+const http2 = require("node:http2");
+
+const server = http2.createServer();
+server.on("error", (err) => console.error(err));
+
+server.on("stream", (stream, headers) => {
+  // stream is a Duplex
+  stream.respond({
+    "content-type": "text/html; charset=utf-8",
```
Comment (KristopherPaulsen) [inline examples/ngrok-http2.js:10]: Looks like a small formatting issue here 👀

**Review by KristopherPaulsen — APPROVED** [review]
Seems good to me, lets ship it.

File: examples/ngrok-http2.js (hunk 1 above)
Comment (OfTheDelmer) [reply examples/ngrok-http2.js:10]: Yeah, its a verbatim copy from Node `http2` docs `https://nodejs.org/api/http2.html#server-side-example` and I feel inclined to let it be.

**Pull request merged**